src/cargo/core/package.rs

```diff
@@ -154,7 +158,7 @@ impl PackageSet {
     pub fn sort(&self) -> Option<PackageSet> {
         let mut graph = graph::Graph::new();
 
-        for pkg in self.packages.iter() {
+        for pkg in &self.packages {
```

Comment:
I personally prefer `for foo in bar.iter()` over `for foo in &bar`, so could these be backed out?